### PR TITLE
Do not codesign transitive dependency iOS pod resource bundles

### LINKED ISSUE #111475: Signing errors on iOS pod bundle resources on Xcode 14 "Signing for "x" requires a development team."
Since I upgraded to Xcode 14, my Flutter project doesn't run because of Signing issues of iOS pods

Error:
```console
Could not build the precompiled application for the device.
Error (Xcode): Signing for "DKPhotoGallery-DKPhotoGallery" requires a development team. Select a development team in the Signing & Capabilities editor.
{path to project folder}/ios/Pods/Pods.xcodeproj

Error (Xcode): Signing for "DKImagePickerController-DKImagePickerController" requires a development team. Select a development team in the Signing & Capabilities editor.
{path to project folder}/ios/Pods/Pods.xcodeproj

Error (Xcode): Signing for "TOCropViewController-TOCropViewControllerBundle" requires a development team. Select a development team in the Signing & Capabilities editor.
{path to project folder}/ios/Pods/Pods.xcodeproj
```

<details><summary>My project's dependencies (to reproduce the problem, the error is from one of these dependencies and the corresponding iOS Pod)</summary>

```yaml
dependencies:
  flutter:
    sdk: flutter

  # TODO: Switch back to main package when this is merged: https://github.com/jhomlala/betterplayer/pull/1059
  better_player:
    git:
      url: https://github.com/matthewrice345/betterplayer.git
      ref: Issue1057
  blur: 3.1.0
  bugsnag_flutter: 2.2.0
  cached_network_image: 3.2.2
  connectivity_plus: 2.3.6+1
  # Needed for tus_client
  cross_file: 0.3.3+1
  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: 1.0.5
  device_info_plus: 4.1.2
  dio: 4.0.6
  easy_localization: 3.0.1
  expandable: 5.0.1
  fading_edge_scrollview: 3.0.0
  faker: 2.0.0
  file_picker: 5.0.1
  firebase_analytics: 9.3.4
  firebase_core: 1.22.0
  firebase_messaging: 13.0.1
  flare_flutter: 3.0.2
  flinq: 2.0.2
  flutter_app_badger: 1.4.0
  flutter_facebook_auth: 4.4.1
  flutter_inappwebview: 5.4.3+7
  flutter_keyboard_visibility: 5.3.0
  flutter_local_notifications: 9.9.1
  flutter_polygon_clipper: 1.0.4
  flutter_slidable: 2.0.0
  flutter_spinkit: 5.1.0
  flutter_staggered_grid_view: 0.6.2
  flutter_svg: 1.1.5
  fluttertoast: 8.0.9
  font_awesome_flutter: 10.2.1
  form_field_validator: 1.1.0
  google_maps_flutter: 2.2.0
  html: 0.15.0
  html_unescape: 2.0.0
  http: 0.13.5
  http_parser: 4.0.1
  image_cropper: 3.0.0
  image_downloader: 0.31.0
  image_picker: 0.8.5+3
  intl: 0.17.0
  json_annotation: 4.6.0
  jwt_decoder: 2.0.1
  mime: 1.0.2
  # TODO: Remove this dependency if it is added to the Flutter core: https://github.com/flutter/flutter/pull/67515
  modal_bottom_sheet: 2.1.2
  # Version checking
  package_info_plus: 1.4.3+1
  path: 1.8.2
  path_provider: 2.0.11
  percent_indicator: 4.2.2
  photo_view: 0.14.0
  pin_code_fields: 7.4.0
  provider: 6.0.3
  pusher_client:
    git:
      url: https://github.com/appcapsergen/pusher_client
  scroll_to_index: 3.0.1
  shared_preferences: 2.0.15
  shimmer: 2.0.0
  sign_in_with_apple: 4.1.0
  timeago: 3.3.0
  tus_client: 1.0.2
  uni_links: 0.5.1
  url_launcher: 6.1.5
  uuid: 3.0.6
  video_player: 2.4.7

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_launcher_icons: 0.10.0
  json_serializable: 6.3.1
  linter: 1.28.0
```
</details>

<details><summary>flutter doctor -v:</summary>

```
[✓] Flutter (Channel stable, 3.3.1, on macOS 12.6 21G115 darwin-arm, locale en-NL)
    • Flutter version 3.3.1 on channel stable at {path to Flutter folder}
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4f9d92fbbd (7 days ago), 2022-09-06 17:54:53 -0700
    • Engine revision 3efdf03e73
    • Dart version 2.18.0
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/{user}/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Users/{user}/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/{user}/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/{user}/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.71.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

### PR DESCRIPTION
**Pull request by jmagman**
Bundles can contain resources like images and localization files.  They are embedded in the framework and should not be codesigned. Xcode 14 changed behavior and is erroneously codesigning these.  CocoaPods has not yet added a workaround on their side https://github.com/CocoaPods/CocoaPods/issues/11402.

Confirms this also works for add-to-app hosts.

Fixes https://github.com/flutter/flutter/issues/111475

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 1]
```diff
@@ -18,6 +18,15 @@ import 'package:path/path.dart' as path;
 /// adding Flutter to an existing iOS app.
 Future<void> main() async {
   await task(() async {
+    // Update pod repo.
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/module_test_ios.dart:21]:
```
[module_test_ios] [STDOUT] stdout:                [!] CocoaPods could not find compatible versions for pod "GoogleSignIn":
[module_test_ios] [STDOUT] stdout:                  In Podfile:
[module_test_ios] [STDOUT] stdout:                    google_sign_in_ios (from `.symlinks/plugins/google_sign_in_ios/ios`) was resolved to 0.0.1, which depends on
[module_test_ios] [STDOUT] stdout:                      GoogleSignIn (~> 6.2)
[module_test_ios] [STDOUT] stdout: 
[module_test_ios] [STDOUT] stdout:                None of your spec sources contain a spec satisfying the dependency: `GoogleSignIn (~> 6.2)`.
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8802907618391831937/+/u/run_module_test_ios/test_stdout

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 2]
```diff
@@ -297,7 +309,7 @@ end
             'CODE_SIGNING_REQUIRED=NO',
             'CODE_SIGN_IDENTITY=-',
             'EXPANDED_CODE_SIGN_IDENTITY=-',
-            'CONFIGURATION_BUILD_DIR=${objectiveCBuildDirectory.path}',
+            'BUILD_DIR=${objectiveCBuildDirectory.path}',
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/module_test_ios.dart:312]: `BUILD_DIR` is used to construct `CONFIGURATION_BUILD_DIR`, as well as a few other build settings that were required to move the resource bundle to the right place.

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 3]
```diff
@@ -18,6 +18,15 @@ import 'package:path/path.dart' as path;
 /// adding Flutter to an existing iOS app.
 Future<void> main() async {
   await task(() async {
+    // Update pod repo.
+    await eval(
+      'pod',
+      <String>['repo', 'update'],
+      environment: <String, String>{
+        'LANG': 'en_US.UTF-8',
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/module_test_ios.dart:26]: Out of curiosity, what kind of bad things could happen if we don't override the LANG?

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 4]
```diff
@@ -201,6 +211,8 @@ dependencies:
       }
 
       checkFileExists(path.join(ephemeralIOSHostApp.path, 'Frameworks', 'url_launcher_ios.framework', 'url_launcher_ios'));
+      // Resources should be embedded.
+      checkDirectoryExists(path.join(ephemeralIOSHostApp.path, 'Frameworks', 'GoogleSignIn.framework', 'GoogleSignIn.bundle'));
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/module_test_ios.dart:215]: The resources are localized string files, btw

File: dev/devicelab/bin/tasks/module_test_ios.dart [hunk 5]
```diff
@@ -306,27 +318,33 @@ end
         );
       });
 
-      final bool existingAppBuilt = exists(File(path.join(
+      final String hostAppDirectory = path.join(
         objectiveCBuildDirectory.path,
+        'Debug-iphoneos',
```
Comment (jmagman) [inline dev/devicelab/bin/tasks/module_test_ios.dart:323]: `BUILD_DIR` doesn't include `Debug-iphoneos` so now it's being added.

File: dev/devicelab/bin/tasks/module_test_ios.dart (hunk 2 above)
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/module_test_ios.dart:312]: whut?

Comment (christopherfujino) [reply dev/devicelab/bin/tasks/module_test_ios.dart:312]: Oh, I asked what this was right after you commented, thanks

Comment (jmagman) [reply dev/devicelab/bin/tasks/module_test_ios.dart:312]: I should have submitted them before requesting review.  You're too speedy 🙂

File: dev/devicelab/bin/tasks/module_test_ios.dart (hunk 1 above)
Comment (christopherfujino) [reply dev/devicelab/bin/tasks/module_test_ios.dart:21]: Thanks. I should start inline commenting failing builds on my fix PRs.

File: dev/devicelab/bin/tasks/module_test_ios.dart (hunk 3 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/module_test_ios.dart:26]: A harmless warning  #45710 https://github.com/CocoaPods/CocoaPods/issues/9347
The tool does this
https://github.com/flutter/flutter/blob/8074811c2f1ec801aff9487e4ba7f49ab540bbd3/packages/flutter_tools/lib/src/macos/cocoapods.dart#L121
but this is invoked outside the tool.  The fewer avoidable warnings in the logs the better, imo.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: dev/devicelab/bin/tasks/module_test_ios.dart (hunk 3 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/module_test_ios.dart:26]: Never mind, I fixed this at some point, `exec`, `eval` are getting `LANG` added later: #82308

**Comment by radvansky-tomas** [comment]
ETA to release this to stable channel ?

**Comment by jmagman** [comment]
Soon, this has already merged into the next hotfix branch: https://github.com/flutter/flutter/pull/111759

**Pull request merged**